tests/integration_tests/build/test_build.py

```diff
@@ -2,60 +2,47 @@
 # SPDX-License-Identifier: Apache-2.0
 """Tests that check if both the debug and the release builds pass."""
 
+import itertools
 import os
+import pytest
 
 import host_tools.cargo_build as host  # pylint:disable=import-error
 
-
-CARGO_DEBUG_REL_PATH = os.path.join(host.CARGO_BUILD_REL_PATH, 'debug')
-CARGO_DEBUG_REL_PATH_FEATURES = os.path.join(
-    host.CARGO_BUILD_REL_PATH,
-    'debug-features'
-)
-CARGO_RELEASE_REL_PATH_FEATURES = os.path.join(
-    host.CARGO_BUILD_REL_PATH,
-    'release-features'
-)
+FEATURES = ["", "vsock"]
+BUILD_TYPES = ["debug", "release"]
 
 
-def test_build_debug(test_session_root_path):
-    """Test if a debug-mode build works."""
-    build_path = os.path.join(
-        test_session_root_path,
-        CARGO_DEBUG_REL_PATH
-    )
-    host.cargo_build(build_path)
-
-
-def test_build_debug_with_features(test_session_root_path):
-    """Test if a debug-mode build works for supported features."""
-    build_path = os.path.join(
-        test_session_root_path,
-        CARGO_DEBUG_REL_PATH_FEATURES
+@pytest.mark.parametrize(
+    "features, build_type",
+    itertools.product(FEATURES, BUILD_TYPES)
+)
+def test_build(test_session_root_path, features, build_type):
+    """
+    Test build using a cartesian product of possible features and build
+    types.
+    """
+    extra_args = ""
+
+    if build_type == "release":
+        extra_args += "--release "
+
+    # The relative path of the binaries is computed using the build_type
+    # (either release or debug) and if any features are provided also using
+    # the features names.
+    # For example, a default release build with no features will end up in
+    # the relative directory "release", but for a vsock release build the
+    # relative directory will be "release-vsock".
```

Comment:
What happens if we parametrize for more than one feature at a time? Would `rel_path` still be valid?

Reply:
The code as is does not support multiple features (mostly because I thought it was very unlikely for us to add any more). But I can implement support for more features as well.